pkg/reconciler/webhook/controller.go

```diff
@@ -0,0 +1,79 @@
+package webhook
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/cache"
+	kubeclient "knative.dev/pkg/client/injection/kube/client"
+	vwhinformer "knative.dev/pkg/client/injection/kube/informers/admissionregistration/v1/validatingwebhookconfiguration"
+	"knative.dev/pkg/controller"
+	secretinformer "knative.dev/pkg/injection/clients/namespacedkube/informers/core/v1/secret"
+	"knative.dev/pkg/logging"
+	pkgreconciler "knative.dev/pkg/reconciler"
+	"knative.dev/pkg/system"
+	"knative.dev/pkg/webhook"
+)
+
+func NewAdmissionController(ctx context.Context,
+	name, path string,
+	wc func(context.Context) context.Context,
+	disallowUnknownFields bool,
+) *controller.Impl {
+
+	client := kubeclient.Get(ctx)
+	vwhInformer := vwhinformer.Get(ctx)
+	secretInformer := secretinformer.Get(ctx)
+	options := webhook.GetOptions(ctx)
+
+	key := types.NamespacedName{
+		Namespace: system.Namespace(),
+		Name:      name,
+	}
+
+	c := &reconciler{
+		LeaderAwareFuncs: pkgreconciler.LeaderAwareFuncs{
+			// Have this reconciler enqueue our singleton whenever it becomes leader.
+			PromoteFunc: func(bkt pkgreconciler.Bucket, enq func(pkgreconciler.Bucket, types.NamespacedName)) error {
+				enq(bkt, key)
+				return nil
+			},
+		},
+
+		key:  key,
+		path: path,
+
+		withContext:           wc,
+		disallowUnknownFields: disallowUnknownFields,
+		secretName:            options.SecretName,
+
+		client:       client,
+		vwhlister:    vwhInformer.Lister(),
+		secretlister: secretInformer.Lister(),
+	}
+
+	logger := logging.FromContext(ctx)
+	cont := controller.NewContext(ctx, c, controller.ControllerOptions{WorkQueueName: "ValidatingWebhook", Logger: logger})
+
+	// Reconcile when the named ValidatingWebhookConfiguration changes.
+	if _, err := vwhInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: controller.FilterWithName(name),
+		// It doesn't matter what we enqueue because we will always Reconcile
+		// the named MWH resource.
+		Handler: controller.HandleAll(cont.Enqueue),
+	}); err != nil {
+		logger.Panicf("Couldn't register ValidatingWebhookConfiguration informer event handler: %w", err)
```

Comment:
error messages in Go should start with a lowercase letter and should contain no trailing punctuation
https://staticcheck.io/docs/checks#ST1005